symfony/plugins/orangehrmTimePlugin/Api/CustomerAPI.php

```diff
@@ -0,0 +1,238 @@
+
+    public const FILTER_NAME = 'name';
+
+    /**
+     * @var CustomerService|null
+     */
+    protected ?CustomerService $customerService = null;
+
+    /**
+     *
+     * @return CustomerService
+     */
+    public function getCustomerService(): CustomerService
+    {
+        if (!$this->customerService instanceof CustomerService) {
+            $this->customerService = new CustomerService();
+        }
+        return $this->customerService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $customerSearchParamHolder = new CustomerSearchFilterParams();
+        $this->setSortingAndPaginationParams($customerSearchParamHolder);
+        $customerSearchParamHolder->setName(
+            $this->getRequestParams()->getStringOrNull(RequestParams::PARAM_TYPE_QUERY, self::FILTER_NAME)
+        );
+        $customers = $this->getCustomerService()->searchCustomers($customerSearchParamHolder);
+        $count = $this->getCustomerService()->getCustomersCount($customerSearchParamHolder);
+
+        return new EndpointCollectionResult(
+            CustomerModel::class, $customers,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::FILTER_NAME),
+            ...$this->getSortingAndPaginationParamsRules(CustomerSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $customer = new Customer();
+        $customerName = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $customerDescription = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_DESCRIPTION
+        );
+        $customer->setName($customerName);
+        $customer->setDescription($customerDescription);
+        $customer->setDeleted(false);
+        $this->getCustomerService()
+            ->getCustomerDao()
+            ->saveCustomer($customer);
+
+        return new EndpointResourceResult(CustomerModel::class, $customer);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_NAME,
+                new Rule(Rules::STRING_TYPE),
+                new Rule(Rules::REQUIRED),
+                new Rule(Rules::LENGTH, [null, self::PARAM_RULE_NAME_MAX_LENGTH])
+            ),
+            new ParamRule(
+                self::PARAMETER_DESCRIPTION,
+                new Rule(Rules::STRING_TYPE),
+                new Rule(Rules::LENGTH, [null, self::PARAM_RULE_DESCRIPTION_MAX_LENGTH])
+            )
```

Comment:
separate it to method since used in `getValidationRuleForUpdate` as well. refer `OrangeHRM\Admin\Api\JobTitleAPI::getCommonBodyValidationRules`

Reply:
done